packages/ec-core/src/operator/selector/mod.rs

```diff
@@ -13,13 +13,64 @@ pub mod worst;
 
 pub use error::EmptyPopulation;
 
+/// Select an individual from a `Population`.
+///
+/// See [`Select`] for a wrapper that converts a `Selector` into an
+/// [`Operator`], allowing selectors to be used in chains of operators.
+///
+/// # Examples
+///
+/// In this example we use the `[Best]` selector to choose the
+/// "best" (maximal) value from a list.
+///
+/// ```
+/// # use ec_core::operator::selector::{best::Best, Selector};
+/// # use rand::thread_rng;
+/// #
+/// let population = vec![5, 8, 9, 2, 3, 6];
```

Comment:
do we need a vec here?

Reply:
We currently _only_ implement the `Population` trait for `Vec<I>` and not for anything else. I feel like (a) there should probably be other implementations but (b) that's not really part of this PR. So I'm going to make another issue and we can come back to that later.